server/src/interfaces/library.interface.ts

```diff
@@ -6,10 +8,10 @@ export const ILibraryRepository = 'ILibraryRepository';
 export interface ILibraryRepository {
   getAll(withDeleted?: boolean): Promise<LibraryEntity[]>;
   getAllDeleted(): Promise<LibraryEntity[]>;
-  get(id: string, withDeleted?: boolean): Promise<LibraryEntity | null>;
-  create(library: Partial<LibraryEntity>): Promise<LibraryEntity>;
+  get(id: string, withDeleted?: boolean): Promise<LibraryEntity | undefined>;
+  create(library: Insertable<Libraries>): Promise<LibraryEntity>;
   delete(id: string): Promise<void>;
   softDelete(id: string): Promise<void>;
-  update(library: Partial<LibraryEntity>): Promise<LibraryEntity>;
+  update(library: Updateable<Libraries> & { id: string }): Promise<LibraryEntity>;
```

Comment:
The other repos use id, update as the signature now. Maybe you can change your to match their pattern